### PR TITLE
fix(label): RTL fix for label margin

### PR DESCRIPTION
**Pull request by sijav**
#### Short description of what this resolves:
RTL fix (scss changes) for label margin for wp, ios and android platform

#### Changes proposed in this pull request:

- Add label margin prop for RTL (md, ios)
- Fix label margin for RTL (md, ios, wp)

**Ionic Version**: 2.x / 3.x

**Fixes**: #11211

### CODE REVIEW

File: src/components/label/label.ios.scss [hunk 1]
```diff
@@ -6,6 +6,9 @@
 /// @prop - Margin of the label
 $label-ios-margin:             $item-ios-padding-top ($item-ios-padding-right / 2) $item-ios-padding-bottom 0 !default;
 
+/// @prop - Margin of the label for RTL
+$label-ios-margin-rtl:             $item-ios-padding-top 0 $item-ios-padding-bottom ($item-ios-padding-right / 2) !default;
```
Comment (manucorporat) [inline src/components/label/label.ios.scss]: Can you align the content with the rest of variables?

File: src/components/label/label.wp.scss [hunk 2]
```diff
@@ -17,6 +17,10 @@ $label-wp-text-color-focused:         color($colors-wp, primary) !default;
   margin: $item-wp-padding-top ($item-wp-padding-right / 2) $item-wp-padding-bottom 0;
 }
 
+[dir="rtl"] .label-wp {
+  margin: $item-wp-padding-top 0 $item-wp-padding-bottom ($item-wp-padding-right / 2);
```
Comment (manucorporat) [inline src/components/label/label.wp.scss]: I know we missed the `$label-wp-margin-rtl`, would you mind adding both?

**Comment by AmitMY** [comment]
Looks good to me. @manucorporat ?

**Comment by brandyscarney** [comment]
Is there (or will there be) a PR for the input as well? This PR messes up the alignment for floating labels since the input's margin is incorrect. Screenshot before this PR:

screen shot 2017-04-24 at 12 13 13 pm

After this PR:

screen shot 2017-04-24 at 12 13 04 pm

**Comment by AmitMY** [comment]
@brandyscarney There is something weird here.
Clearly, the margin left is `($item-md-padding-right / 2)`, but you screenshot shows 0. 

Also, no, there is no current input rtl PR, all in good time :)

**Comment by brandyscarney** [comment]
@AmitMY This rule is taking precedence so it is getting `margin-left: 0`

screen shot 2017-04-24 at 12 53 14 pm

**Comment by sijav** [comment]
@brandyscarney I'm doing it now, looks like one pr should be for input and another one for select and I think search-bar must change too ...

**Comment by AmitMY** [comment]
@sijav Take a look at [this](https://github.com/driftyco/ionic/pull/11342/files#diff-6ca9a05dab013b2e6056eea0e0b9da6aR304).
If it will get merged, then we can do the same thing for margin, border, and position, and to fix all of the components together, once. We are still finalising the syntax and changes, but I think this (or some version of it) will be merged soon.

I'm letting you know, just so you don't do extra work that will be rejected if this goes in. I will keep you updated

**Comment by sijav** [comment]
@AmitMY Great job! Thanks for letting me know, but we still need some support from typescript (searchbar for ios or sliding item component and so on)

**Comment by AmitMY** [comment]
@brandyscarney This can be closed

**Comment by sijav** [comment]
Closing this PR in favor of [#11342](https://github.com/driftyco/ionic/pull/11342/files#diff-6ca9a05dab013b2e6056eea0e0b9da6aR304)

**Pull request closed without merging**